feat: Module release meta watch (#1968)
kyma-project/lifecycle-manager: docs/contributor/02-controllers.md

The Kyma CR in Kyma Control Plane shows the initial specification and the current status. To install a module, Lifecycle Manager uses the specification from the remote cluster Kyma CR.

### Requeuing the Kyma CR

> The `Kyma` CR is requeued at set intervals using specific flags from the Lifecycle Manager. This ensures that the Kyma CR is periodically reprocessed, allowing the controller to detect and apply any changes that may have occurred during that time. Additionally, several watch mechanisms have been implemented, enabling the controller to requeue Kyma CRs when certain events occur.

Reviewer: ```suggestion
The `Kyma` CR is requeued at set intervals using specific flags from Lifecycle Manager. The requeuing ensures that the Kyma CR is periodically reprocessed, allowing the controller to detect and apply any changes that may have occurred during that time. Additionally, several watch mechanisms are implemented, enabling the controller to requeue Kyma CRs when certain events occur.
```

> These watches monitor `Kyma`, `Secret`, `Manifest`, and `ModuleReleaseMeta` CRs, ensuring that the relevant Kyma CRs are requeued whenever these CRs are created, updated, or deleted. In particular, the watch mechanism for `ModuleReleaseMeta` CRs has a [dedicated implementation](../../internal/watch/modulereleasemeta_change.go). This ensures that any Kyma CRs using a module in a channel affected by the `ModuleReleaseMeta` CR are requeued as needed.

Reviewer: ```suggestion
These watch mechanisms monitor Kyma, Secret, Manifest, and ModuleReleaseMeta CRs, ensuring that the relevant Kyma CRs are requeued whenever these CRs are created, updated, or deleted. Additionally, the watch mechanism for ModuleReleaseMeta CRs has a [dedicated implementation](../../internal/watch/modulereleasemeta_change.go), which ensures that all Kyma CRs using a module in a channel affected by the ModuleReleaseMeta CR are requeued as needed.
```
